### PR TITLE
Join Fixes and Resolver checks for hustle support

### PR DESCRIPTION
**Pull request by srsuryadev**
## Summary
* Fixes in Join to support join on non-unique columns (Right now only for single Join - will extend in the upcoming PRs)
* Resolver fixes to support queries without aggregate or join operator
* Fixes in the installation script.
* removed unused modules
* Simplified the sql test files
* Fixes the test data generator file and fixes in the corresponding sql test files.
* Fixes in the resolver for handling queries that are unsupported constructs (HAVING, IN, EXISTS, etc) and Nested subqueries and also included corresponding tests

### CODE REVIEW

File: src/operators/join/join.cc [hunk 1]
```diff
@@ -253,17 +262,29 @@ void Join::ProbeHashTableBlock(
...
           auto key_value_pair =
               hash_tables_[join_id]->find(left_join_chunk_data[row]);
           if (key_value_pair != hash_table_end) {
-            joined_left_indices[num_joined_indices] = row + offset;
-            joined_right_indices[num_joined_indices] =
-                key_value_pair->second.index;
-            ++num_joined_indices;
+            for (auto record_id : key_value_pair->second) {
+              if (indices_length <= num_joined_indices) {
+                indices_length << 1;
```
Comment (GindaChen) [inline src/operators/join/join.cc:278]: `indices_length <<= 1;` ? Do you mean to shift the `indicies_length`?

File: src/operators/join/join.cc [hunk 2]
```diff
@@ -272,10 +293,18 @@ void Join::ProbeHashTableBlock(
         auto key_value_pair =
             hash_tables_[join_id]->find(left_join_chunk_data[row]);
         if (key_value_pair != hash_table_end) {
-          joined_left_indices[num_joined_indices] = row + offset;
-          joined_right_indices[num_joined_indices] =
-              key_value_pair->second.index;
-          ++num_joined_indices;
+          for (auto record_id : key_value_pair->second) {
+            if (indices_length <= num_joined_indices) {
+              indices_length << 1;
```
Comment (GindaChen) [inline src/operators/join/join.cc:298]: Shift indicies?

**Review by GindaChen — APPROVED** [review]
Awesome PR! Just a few little comments

File: src/operators/join/join.cc (hunk 1 above)
Comment (srsuryadev) [reply src/operators/join/join.cc]: Oop! Thank you!

**Pull request merged**